java/vector/src/main/java/org/apache/arrow/vector/ipc/WriteChannel.java

```diff
@@ -62,15 +66,22 @@ public long write(byte[] buffer) throws IOException {
     return write(ByteBuffer.wrap(buffer));
   }
 
+  long write(byte[] buffer, int offset, int length) throws IOException {
+    return write(ByteBuffer.wrap(buffer, offset, length));
+  }
+
   /**
    * Writes <zeroCount>zeroCount</zeroCount> zeros the underlying channel.
    */
   public long writeZeros(long zeroCount) throws IOException {
     long bytesWritten = 0;
-    while (zeroCount > 0) {
-      int bytesToWrite = (int)Math.min(zeroCount, Integer.MAX_VALUE);
-      bytesWritten += write(new byte[bytesToWrite]);
-      zeroCount -= Integer.MAX_VALUE;
+    long wholeWordsEnd = zeroCount - 8;
+    while (bytesWritten <= wholeWordsEnd) {
+      bytesWritten += write(zeroBytes);
```

Comment:
Are you sure that's a good idea? You're doing many small `write` calls for 8 bytes each time. Unless `WritableByteChannel` is a trivial class, that sounds wasteful. 

Since Arrow recommends 64-byte alignment, I think you also want to make `zeroBytes` 64 bytes long to minimize the number of `write` calls.

Reply:
cc @emkornfield

Reply:
Ok, it seems the `writeZeros` call in `MessageSerializer` should never write more than 8 bytes. So it seems this should be ok :-)

Reply:
Thank you for the good suggestions.